pmd-java/src/main/resources/category/java/codestyle.xml

```diff
@@ -2135,7 +2135,8 @@ E.g. `int[] x = new int[] { 1, 2, 3 };` can be written as `int[] x = { 1, 2, 3 }
         <properties>
             <property name="xpath">
                 <value><![CDATA[
-//VariableDeclarator
+//(LocalVariableDeclaration[@TypeInferred = false()] | FieldDeclaration)
```

Comment:
I think you may simplify this by reverting to `//VariableDeclarator` here and instead using VariableDeclaratorId's own `@TypeInferred` attr below, which independently of field or local var context gives the correct result ^^